solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -114,4 +114,65 @@ print(ans)
 ```
 
 </PySection>
+
+	
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class GCDBoard {
+    static final int maxN = 100000 + 5;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        int N = Integer.parseInt(br.readLine());
+        StringTokenizer st = new StringTokenizer(br.readLine());
+
+        int[] arr = new int[maxN];
+        for (int x = 1; x <= N; x++) {
+		arr[x] = Integer.parseInt(st.nextToken());
+        }
+
+        int[] prefix = new int[maxN];
+	// Prefix will find the GCD from the start
+        int[] suffix = new int[maxN];
+	// Suffix will find the GCD from the end
+
+        prefix[0] = 0;
+        suffix[N + 1] = 0;
+
+        for (int x = 1; x <= N; ++x) {
+		// GCD for prefix
+		prefix[x] = gcd(prefix[x - 1], arr[x]);
+        }
+        for (int x = N; x >= 1; --x) {
+		// GCD for suffix
+		suffix[x] = gcd(suffix[x + 1], arr[x]);
+        }
+
+        int ans = 1;
+
+        for (int x = 1; x <= N; ++x) {
+		// 'x' is the index we are going to remove
+		ans = Math.max(ans, gcd(prefix[x - 1], suffix[x + 1]));
+		// Removing index 'x is the same as finding the GCD of everything
+		// before it and after it
+        }
+
+        System.out.println(ans);
+    }
+
+    public static int gcd(int num1, int num2) {
+	// How to quickly find GCD
```

Comment:
same here.

Reply:
Function indentation might look a little wonky, but could you check please?